app/lti.go

```
@@ -131,3 +130,27 @@ func (a *App) addTeamMemberIfRequired(userId string, teamName string) *model.App
 
 	return nil
 }
+
+func (a *App) GetUserByLTI(ltiUserID string) (*model.User, *model.AppError) {
+	if result := <-a.Srv.Store.User().GetByLTI(ltiUserID); result.Err != nil && result.Err.Id == "store.sql_user.get_by_lti.missing_account.app_error" {
+		result.Err.StatusCode = http.StatusNotFound
+		return nil, result.Err
+	} else if result.Err != nil {
+		result.Err.StatusCode = http.StatusBadRequest
+		return nil, result.Err
+	} else {
+		return result.Data.(*model.User), nil
+	}
+}
+
+// GetLTIUser can be used to get an LTI user by lti user id or email
+func (a *App) GetLTIUser(ltiUserID, email string) (*model.User, *model.AppError) {
+	user, err := a.GetUserByLTI(ltiUserID)
+	if err != nil {
+		user, err = a.GetUserByEmail(email)
+	}
+	if err != nil {
+		return nil, err
```

I don't think we should return the original error here. What that would do is present the scenario that user couldn't be found by email. Also, notice that only one of the error object is returned - from `GetUserByEmail` and not from `GetUserByLTI`.

I think what we should do here is create a new error object that just represents the scenario of an LTI user not being found.

---

Updated the method as discussed.